### DISCUSSION TITLE
Combining multi-line constructor args in to single line

### DISCUSSION THREAD

**Discussion by benr77** [PHP-CS-Fixer/PHP-CS-Fixer#6820] [Q&A]
## Support question

Is there a rule to combine multi-line constructor arguments in to a single line, when it doesn't violate a line length policy? I would like this to apply only to method definitions not method calls as well.

e.g.

```php
public function __construct(
    Request $request,
    Property $property
): JsonResponse {
```

should be fixed to 

```php
public function __construct(Request $request, Property $property): JsonResponse {
```

I ask because a different fixer will change a constructor argument line that is very long in to a multi-line call, but if e.g. you rename some parameters and it would then fit back on to a single line I don't want to have to make this reverse change manually.

Thank you.

**Comment by benr77** [comment]
For reference I have found this fixer
https://github.com/symplify/coding-standard/blob/master/docs/rules_overview.md#linelengthfixer

**Reply by Wirone** [reply]
If #7064 is merged then it would be possible to use Symplify's rules in native Fixer (not ECS), of course with some custom setup because these rules require some params in the constructor. If you like this addition, then vote/comment on the PR 🙂.
